/*
 * This file was automatically generated by EvoSuite
 * Tue Feb 23 18:01:29 GMT 2021
 */

package lp2.projetofinal.controllers;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import lp2.projetofinal.controllers.Sistema;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class Sistema_ESTest extends Sistema_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test00()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.listarEmprestimosUsuarioPegandoEmprestado("r", "~('F[S?");
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test01()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.listarEmprestimosUsuarioEmprestando("z<K8", "Emprestimos: ");
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test02()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.devolverItem("*;I*ICN(m@ONk`I(]", "hC(p*qP@8yE]", "*;I*ICN(m@ONk`I(]", "Y)7+y$9*~7kn3qv@D", "*;I*ICN(m@ONk`I(]", "Y)7+y$9*~7kn3qv@D", "M");
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test03()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.registrarEmprestimo("Emprestimos: ", "2?q:>.kd#QX-", "2?q:>.kd#QX-", "KSlfe`/OYWLl'eW", "KSlfe`/OYWLl'eW", "KSlfe`/OYWLl'eW", 990);
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test04()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.registrarEmprestimo("?%XrpgBx DWf39:7R", "lp2.projetofinal.controllers.Sistema", "", "", (String) null, "g.thNe'H~=m]=;v;4", (-1223));
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Nome usuario nao pode ser vazio ou nulo
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test05()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.getInfoItem("tP&q9O", "tP&q9O", "9", "lp2.projetofinal.controllers.Sistema");
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test06()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.atualizarItem("i7lb@,@", "2v[rmOno,hc{", "lp2.projetofinal.orders.OrdenaUsuarioReputacaoDecrescente", "i7lb@,@", "i7lb@,@");
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test07()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.atualizarItem("-3Nqog=qwYV*[L]+V-", "-3Nqog=qwYV*[L]+V-", "Ob[mmgVV-", "-3Nqog=qwYV*[L]+V-", "R");
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test08()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.removerItem("dSst$93yJ)@:]8b5", "zh?=-AFQ.", "Emprestimos pegos: ");
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test09()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.adicionarPecaPerdida("I", "{jt2x/2I~/P;PdMw+", "p", "B[nVG3E9Q,A^`7");
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test10()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.adicionarBluRay("@C`K_$?_ns", "oJ", "oJ", 1);
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test11()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.atualizarUsuario("lp2.projetofinal.controllers.Sistema", "._y`fL`h^", "lp2.projetofinal.controllers.Sistema", "lp2.projetofinal.controllers.Sistema");
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test12()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      sistema0.cadastrarUsuario("41o,^$X2^#b", "/'`!.", ",M");
  }

  @Test(timeout = 4000)
  public void test13()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      sistema0.cadastrarUsuario("t/N 8MAyWz", "t/N 8MAyWz", "t/N 8MAyWz");
      sistema0.removerUsuario("t/N 8MAyWz", "t/N 8MAyWz");
  }

  @Test(timeout = 4000)
  public void test14()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.registrarEmprestimo("K", "K", "K", "<w|4zir,HP|D$3!6(", ",y9V u<]XJM><U$", "K", 130);
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test15()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.getInfoUsuario("/y 'ZN%v@5Z-IE>", "|i?p}J", "Telefone");
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test16()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.getInfoItem("?j", "}}?(=Cbb/", "{6neJtP[uL,O6t", "?j");
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test17()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.cadastrarJogoTabuleiro("5BC1SDZ2>f:q\"jYC", "Bj!", "B", 0.05);
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test18()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.cadastrarEletronico("{}2", "{}2", "{}2", 748.0, "d=-\"Tqc7JJ");
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test19()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.cadastrarBluRayShow("@D3", "m7!WA>NteM", " - Reputacao: ", 2252.6641081972, (-477), (-2147), "", "8<.");
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Duracao invalida
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test20()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.cadastrarBluRayFilme("L95", "L95", "L95", 876.1053526439434, 2690, "vuV|Olp/ 3", "L95", 2424);
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test21()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.atualizarItem("(pJ(qvBc[=%z`1", "(pJ(qvBc[=%z`1", "Q>x8i+vX4lCTZv", "TFI", "TFI");
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test22()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.adicionarBluRay("J+RwTZ.FbK", "/Jy'N{xEh,g7inPL7", "J+RwTZ.FbK", 11);
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test23()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      sistema0.cadastrarUsuario("EwkAKW>+", "EwkAKW>+", "EwkAKW>+");
      String string0 = sistema0.listarEmprestimosUsuarioPegandoEmprestado("EwkAKW>+", "EwkAKW>+");
      assertEquals("Nenhum item pego emprestado", string0);
  }

  @Test(timeout = 4000)
  public void test24()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.removerUsuario("C6-X/P1PH19'aj0u&", "F(ARy=8Q");
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test25()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.removerItem("lp2.projetofinal.controllers.Sistema", "lp2.projetofinal.controllers.Sistema", (String) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test26()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.registrarEmprestimo("t/N 8MAyWz", (String) null, "|8vJ", (String) null, "`&db`a+}Fe\"QHS", "kaP{c|~3[\"bpgNW", (-2241));
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test27()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.pesquisarDetalhesItem("&3On&r`Z%.)rc=lc", (String) null, "");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test28()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.listarEmprestimosUsuarioPegandoEmprestado("lp2.projetofinal.controllers.ControllerUsuario", (String) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test29()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.listarEmprestimosItem((String) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test30()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.listarEmprestimosItem("");
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Nome item nao pode ser vazio ou nulo
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test31()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.getInfoUsuario((String) null, "", "");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test32()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.getInfoItem("X24J`s h &pXS", "Y;,VI,wmC>n*&#", (String) null, "Y;,VI,wmC>n*&#");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test33()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.devolverItem("Nenhum item pego emprestado", "#", "Nenhum emprestimos associados ao item", "Nenhum emprestimos associados ao item", " 0A[a1\"D", "b", " 0A[a1\"D");
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test34()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.cadastrarUsuario(" emprestimos - ", (String) null, " emprestimos - ");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test35()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.cadastrarUsuario("", "", "Reputacao");
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Nome usuario nao pode ser vazio ou nulo
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test36()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.cadastrarJogoTabuleiro("lp2.projetofinal.controllers.Sistema", "lp2.projetofinal.controllers.Sistema", (String) null, 1.0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test37()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.cadastrarEletronico("", "", "", 1669.974641805, "");
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Nome usuario nao pode ser vazio ou nulo
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test38()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.cadastrarBluRayShow((String) null, "IYJNP", "", (-829.0), (-465), 1, (String) null, "IYJNP");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test39()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.cadastrarBluRaySerie("gSTCU<x_u", (String) null, "gSTCU<x_u", 2021.3042, "gSTCU<x_u", 0, "m+'6SagBD;I%:K", (String) null, (-1414));
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test40()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.cadastrarBluRayFilme("?T4~", "n", (String) null, (-165.7), (-1), "", (String) null, 1791);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test41()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.atualizarUsuario("<2B:btTTL9I9z'n/Y", (String) null, "", "(\"z");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test42()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.atualizarItem("Nenhum item pego emprestado", "<!H%;>ZG<", (String) null, "", "<!H%;>ZG<");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test43()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.adicionarPecaPerdida("bh", "bh", (String) null, "lp2.projetofinal.controllers.Sistema");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test44()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      String string0 = "\\u";
      // Undeclared exception!
      try { 
        sistema0.adicionarBluRay(string0, (String) null, "", 186);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test45()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      String string0 = sistema0.listarItensEmprestados();
      assertEquals("", string0);
  }

  @Test(timeout = 4000)
  public void test46()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      String string0 = sistema0.listarItensNaoEmprestados();
      assertEquals("", string0);
  }

  @Test(timeout = 4000)
  public void test47()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.listarEmprestimosUsuarioEmprestando((String) null, "");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test48()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.atualizarUsuario("|", "|", ": ", "lp2.projetofinal.controllers.Sistema");
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test49()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.removerUsuario("Nome", (String) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test50()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.pesquisarDetalhesItem("wiKi\"GGeJJB4", "l", "(,8g");
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test51()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      String string0 = sistema0.listarCaloteiros();
      assertEquals("Lista de usuarios com reputacao negativa: ", string0);
  }

  @Test(timeout = 4000)
  public void test52()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      sistema0.inicializar();
  }

  @Test(timeout = 4000)
  public void test53()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      String string0 = sistema0.listarTop10MelhoresUsuarios();
      assertEquals("", string0);
  }

  @Test(timeout = 4000)
  public void test54()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      sistema0.finalizar();
  }

  @Test(timeout = 4000)
  public void test55()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      String string0 = sistema0.listarEmprestimosItem(" 0A[a1\"D");
      assertEquals("Nenhum emprestimos associados ao item", string0);
  }

  @Test(timeout = 4000)
  public void test56()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      String string0 = sistema0.listarTop10Itens();
      assertEquals("", string0);
  }

  @Test(timeout = 4000)
  public void test57()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      sistema0.resetarPersistencia();
  }

  @Test(timeout = 4000)
  public void test58()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      String string0 = sistema0.listarTop10PioresUsuarios();
      assertEquals("", string0);
  }

  @Test(timeout = 4000)
  public void test59()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      String string0 = sistema0.listarItensOrdenadosPorValor();
      assertEquals("", string0);
  }

  @Test(timeout = 4000)
  public void test60()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      sistema0.cadastrarUsuario("t/N 8MAyWz", "t/N 8MAyWz", "t/N 8MAyWz");
      String string0 = sistema0.listarTop10MelhoresUsuarios();
      assertEquals("1: t/N 8MAyWz - Reputacao: 0.00|", string0);
  }

  @Test(timeout = 4000)
  public void test61()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      String string0 = sistema0.listarItensOrdenadosPorNome();
      assertEquals("", string0);
  }

  @Test(timeout = 4000)
  public void test62()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.cadastrarEletronico("|8vJ", "lp2.projetofinal.persistencia.Persistencia", (String) null, 543.1, "Emprestimos pegos: ");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }

  @Test(timeout = 4000)
  public void test63()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.cadastrarBluRaySerie("16yj ]T.B2]=5Emka", " 0A[a1\"D", "gRY", 100.0, "Nenhum emprestimos associados ao item", 3365, "Zy$", "8(", 3365);
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Usuario invalido
         //
         verifyException("lp2.projetofinal.util.Exceptions", e);
      }
  }

  @Test(timeout = 4000)
  public void test64()  throws Throwable  {
      Sistema sistema0 = new Sistema();
      // Undeclared exception!
      try { 
        sistema0.cadastrarBluRayFilme("X7", "GOs%zKhtnw3", "}", (-3318.412937981), (-1195), "}", "l$~kUy0}b_W/U{", (-1195));
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Preco invalido
         //
         verifyException("lp2.projetofinal.util.Checks", e);
      }
  }
}
